### ISSUE TITLE
The control buttons and shadow of a window with custom title bar is so weird

### ISSUE THREAD

**Issue by hez2010** [microsoft/microsoft-ui-xaml#7131]
### Describe the bug

This is how a window with default system title bar looks like:

![image](https://user-images.githubusercontent.com/14960345/169633047-ce7a5927-0d3c-48e0-bdf7-046c9fb55b12.png)

This is how a window with `ExtendsContentIntoTitleBar=true` looks like:

![image](https://user-images.githubusercontent.com/14960345/169633069-fd21ea0c-d17c-4335-b624-fb4427fa9fcc.png)

You can observe that the size (and the shape) of control buttons is not consistent:

![image](https://user-images.githubusercontent.com/14960345/169633102-1d512588-c0d2-4428-8a59-f5af6ea0e8f7.png)

Also, the depth of shadow of the window is not identical:

![image](https://user-images.githubusercontent.com/14960345/169633234-878498aa-765a-47c2-a4e4-d3fc260bd520.png)

### Steps to reproduce the bug

1. Create a new blank WinUI 3 app with WASDK 1.1-preview3
2. Set `ExtendsContentIntoTitleBar` to `true`

### Expected behavior

The control buttons as well as shadow of the window should be identical both with system title bar and custom title bar.

### NuGet package version

WinUI 3 - Windows App SDK 1.1 Preview 3: 1.1.0-preview3

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows Insider Build (22621)

**Comment by hez2010** [comment]
Does this issue belong to WindowsAppSDK? If so, can someone help me transfer the issue to https://github.com/microsoft/WindowsAppSDK?

**Comment by riverar** [comment]
![](https://user-images.githubusercontent.com/475132/174511030-72c3a73d-7c9e-452e-8682-752e9ae8b521.png)

This still reproduces in Windows App SDK 1.1.1. Only difference between the two windows above is `ExtendsContentIntoTitleBar = true`

Ping @rkarman

**Comment by rkarman** [comment]
Thanks for the ping here @riverar, I was not aware of this issue being logged.

@marb2000 / @MikeHillberg can one of you move this issue over to WinAppSDK so we can assign the proper owners and track this with the windowing team?

**Comment by sigmarsson** [comment]
1. I witness the same with 1.1.4 :

![image](https://user-images.githubusercontent.com/43340959/187520503-3b3970fb-a2b7-4348-ae9b-293272194069.png)

2. Furthermore the caption buttons receive only double mouse clicks and all three buttons operate as Max/Unmax the window. 

3. Furthermore I associate a ```TeextBlock``` to the title bar but never appears on the extended title bar area : 

![image](https://user-images.githubusercontent.com/43340959/187521305-3c76f898-12d5-45e2-88e9-fe387b2be041.png)

**Comment by ritt-app** [comment]
Same behaviour observed for Windows App SDK 1.2.221116.1

**Comment by riverar** [comment]
@MikeHillberg @jonwis Can you assist here with a move to WAS repo?

**Comment by MikeHillberg** [comment]
I think we're in the right place; this looks like [Window.ExtendsContentIntoTitleBar](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.Window.ExtendsContentIntoTitleBar).

@bpulliam

**Comment by RGarrido03** [comment]
Unfortunately this isn’t solved yet in the latest WinAppSDK version. 
However, there must be some trick: the latest Phone Link doesn’t suffer from this bug, yet the WinUI 3 Controls app does. Both are made with WinAppSDK & WinUI 3, of course.

**Comment by sigmarsson** [comment]
Ensure this is invoked after you ```.Activate()``` your window : 

(stumbled over this in WinUI Git Hub example)


```
internal class Win32Ex
{
    private const int WAINACTIVE = 0x00;
    private const int WAACTIVE = 0x01;
    private const int WMACTIVATE = 0x0006;

    [DllImport("user32.dll")]
    private static extern IntPtr GetActiveWindow();

    [DllImport("user32.dll", CharSet = CharSet.Auto)]
    private static extern IntPtr SendMessage(IntPtr hWnd, int msg, int wParam, IntPtr lParam);

    public static void UpdateTitleBar(UIElement shell)
    {
        if (shell is FrameworkElement frameworkElement)
        {
            UpdateTitleBar(frameworkElement.RequestedTheme);
        }
        else
        {
            throw new ArgumentException($"Unable to update application title bar: {shell.GetType()}");
        }
    }

    public static void UpdateTitleBar(ElementTheme theme)
    {
        if (App.Window.ExtendsContentIntoTitleBar)
        {
            if (theme != ElementTheme.Default)
            {
                Application.Current.Resources["WindowCaptionForeground"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Colors.White),
                    ElementTheme.Light => new SolidColorBrush(Colors.Black),
                    _ => new SolidColorBrush(Colors.Transparent)
                };

                Application.Current.Resources["WindowCaptionForegroundDisabled"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
                    ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
                    _ => new SolidColorBrush(Colors.Transparent)
                };

                Application.Current.Resources["WindowCaptionButtonBackgroundPointerOver"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x33, 0xFF, 0xFF, 0xFF)),
                    ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x33, 0x00, 0x00, 0x00)),
                    _ => new SolidColorBrush(Colors.Transparent)
                };

                Application.Current.Resources["WindowCaptionButtonBackgroundPressed"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
                    ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
                    _ => new SolidColorBrush(Colors.Transparent)
                };

                Application.Current.Resources["WindowCaptionButtonStrokePointerOver"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Colors.White),
                    ElementTheme.Light => new SolidColorBrush(Colors.Black),
                    _ => new SolidColorBrush(Colors.Transparent)
                };

                Application.Current.Resources["WindowCaptionButtonStrokePressed"] = theme switch
                {
                    ElementTheme.Dark => new SolidColorBrush(Colors.White),
                    ElementTheme.Light => new SolidColorBrush(Colors.Black),
                    _ => new SolidColorBrush(Colors.Transparent)
                };
            }

            Application.Current.Resources["WindowCaptionBackground"] = new SolidColorBrush(Colors.Transparent);
            Application.Current.Resources["WindowCaptionBackgroundDisabled"] = new SolidColorBrush(Colors.Transparent);

            var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(App.Window);

            if (hwnd == GetActiveWindow())
            {
                SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
                SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
            }
            else
            {
                SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
                SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
            }
        }
    }
}

```

**Comment by RGarrido03** [comment]
> Ensure this is invoked after you ```.Activate()``` your window : 
> 
> 
> 
> (stumbled over this in WinUI Git Hub example)
> 
> 
> 
> 
> 
> ```
> 
> internal class Win32Ex
> 
> {
> 
>     private const int WAINACTIVE = 0x00;
> 
>     private const int WAACTIVE = 0x01;
> 
>     private const int WMACTIVATE = 0x0006;
> 
> 
> 
>     [DllImport("user32.dll")]
> 
>     private static extern IntPtr GetActiveWindow();
> 
> 
> 
>     [DllImport("user32.dll", CharSet = CharSet.Auto)]
> 
>     private static extern IntPtr SendMessage(IntPtr hWnd, int msg, int wParam, IntPtr lParam);
> 
> 
> 
>     public static void UpdateTitleBar(UIElement shell)
> 
>     {
> 
>         if (shell is FrameworkElement frameworkElement)
> 
>         {
> 
>             UpdateTitleBar(frameworkElement.RequestedTheme);
> 
>         }
> 
>         else
> 
>         {
> 
>             throw new ArgumentException($"Unable to update application title bar: {shell.GetType()}");
> 
>         }
> 
>     }
> 
> 
> 
>     public static void UpdateTitleBar(ElementTheme theme)
> 
>     {
> 
>         if (App.Window.ExtendsContentIntoTitleBar)
> 
>         {
> 
>             if (theme != ElementTheme.Default)
> 
>             {
> 
>                 Application.Current.Resources["WindowCaptionForeground"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
> 
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
> 
> 
>                 Application.Current.Resources["WindowCaptionForegroundDisabled"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
> 
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
> 
> 
>                 Application.Current.Resources["WindowCaptionButtonBackgroundPointerOver"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x33, 0xFF, 0xFF, 0xFF)),
> 
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x33, 0x00, 0x00, 0x00)),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
> 
> 
>                 Application.Current.Resources["WindowCaptionButtonBackgroundPressed"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
> 
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
> 
> 
>                 Application.Current.Resources["WindowCaptionButtonStrokePointerOver"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
> 
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
> 
> 
>                 Application.Current.Resources["WindowCaptionButtonStrokePressed"] = theme switch
> 
>                 {
> 
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
> 
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
> 
>                     _ => new SolidColorBrush(Colors.Transparent)
> 
>                 };
> 
>             }
> 
> 
> 
>             Application.Current.Resources["WindowCaptionBackground"] = new SolidColorBrush(Colors.Transparent);
> 
>             Application.Current.Resources["WindowCaptionBackgroundDisabled"] = new SolidColorBrush(Colors.Transparent);
> 
> 
> 
>             var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(App.Window);
> 
> 
> 
>             if (hwnd == GetActiveWindow())
> 
>             {
> 
>                 SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
> 
>                 SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
> 
>             }
> 
>             else
> 
>             {
> 
>                 SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
> 
>                 SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
> 
>             }
> 
>         }
> 
>     }
> 
> }
> 
> 
> 
> ```

Thank you very much for the solution! I'll try that later today

**Comment by RGarrido03** [comment]
> Ensure this is invoked after you `.Activate()` your window :
> 
> (stumbled over this in WinUI Git Hub example)
> 
> ```
> internal class Win32Ex
> {
>     private const int WAINACTIVE = 0x00;
>     private const int WAACTIVE = 0x01;
>     private const int WMACTIVATE = 0x0006;
> 
>     [DllImport("user32.dll")]
>     private static extern IntPtr GetActiveWindow();
> 
>     [DllImport("user32.dll", CharSet = CharSet.Auto)]
>     private static extern IntPtr SendMessage(IntPtr hWnd, int msg, int wParam, IntPtr lParam);
> 
>     public static void UpdateTitleBar(UIElement shell)
>     {
>         if (shell is FrameworkElement frameworkElement)
>         {
>             UpdateTitleBar(frameworkElement.RequestedTheme);
>         }
>         else
>         {
>             throw new ArgumentException($"Unable to update application title bar: {shell.GetType()}");
>         }
>     }
> 
>     public static void UpdateTitleBar(ElementTheme theme)
>     {
>         if (App.Window.ExtendsContentIntoTitleBar)
>         {
>             if (theme != ElementTheme.Default)
>             {
>                 Application.Current.Resources["WindowCaptionForeground"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
> 
>                 Application.Current.Resources["WindowCaptionForegroundDisabled"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
> 
>                 Application.Current.Resources["WindowCaptionButtonBackgroundPointerOver"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x33, 0xFF, 0xFF, 0xFF)),
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x33, 0x00, 0x00, 0x00)),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
> 
>                 Application.Current.Resources["WindowCaptionButtonBackgroundPressed"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Color.FromArgb(0x66, 0xFF, 0xFF, 0xFF)),
>                     ElementTheme.Light => new SolidColorBrush(Color.FromArgb(0x66, 0x00, 0x00, 0x00)),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
> 
>                 Application.Current.Resources["WindowCaptionButtonStrokePointerOver"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
> 
>                 Application.Current.Resources["WindowCaptionButtonStrokePressed"] = theme switch
>                 {
>                     ElementTheme.Dark => new SolidColorBrush(Colors.White),
>                     ElementTheme.Light => new SolidColorBrush(Colors.Black),
>                     _ => new SolidColorBrush(Colors.Transparent)
>                 };
>             }
> 
>             Application.Current.Resources["WindowCaptionBackground"] = new SolidColorBrush(Colors.Transparent);
>             Application.Current.Resources["WindowCaptionBackgroundDisabled"] = new SolidColorBrush(Colors.Transparent);
> 
>             var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(App.Window);
> 
>             if (hwnd == GetActiveWindow())
>             {
>                 SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
>                 SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
>             }
>             else
>             {
>                 SendMessage(hwnd, WMACTIVATE, WAACTIVE, IntPtr.Zero);
>                 SendMessage(hwnd, WMACTIVATE, WAINACTIVE, IntPtr.Zero);
>             }
>         }
>     }
> }
> ```

Thank you very much!! I’ll try that later today :)

**Comment by vanduyquang100** [comment]
The problem is still not fixed yet. 
 But using `Window.AppWindow.TitleBar.ExtendsContentIntoTitleBar = true;` would give the expected result.

**Comment by hez2010** [comment]
BTW in WASDK 1.4, the control buttons issue has been fixed.

**Comment by pratikone** [comment]
closing as fixed in winappsdk 1.4
